presto-hive/src/test/java/com/facebook/presto/hive/hudi/TestHudiIntegration.java

```diff
@@ -0,0 +1,152 @@
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.hive.hudi.HudiTestingDataGenerator.DATA_COLUMNS;
+import static com.facebook.presto.hive.hudi.HudiTestingDataGenerator.HUDI_META_COLUMNS;
+import static com.facebook.presto.hive.hudi.HudiTestingDataGenerator.PARTITION_COLUMNS;
+import static java.lang.String.format;
+
+public class TestHudiIntegration
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HudiTestUtils.createQueryRunner(
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                metastore -> new HivePlugin("hive", metastore),
+                "hive",
+                "hive",
+                "testing");
+    }
+
+    @Test
+    public void testMetadata()
+    {
+        assertQuery("show tables",
+                "SELECT * FROM VALUES " +
+                        "('stock_ticks_cow'), " +
+                        "('stock_ticks_cown'), " +
+                        "('stock_ticks_mor_ro'), " +
+                        "('stock_ticks_mor_rt')," +
+                        "('stock_ticks_morn_ro')," +
+                        "('stock_ticks_morn_rt')");
+
+        FunctionAndTypeManager typeManager = getQueryRunner().getMetadata().getFunctionAndTypeManager();
+
+        // partition tables
+        @Language("SQL") String sql = generateDescribeIdenticalQuery(typeManager, HUDI_META_COLUMNS, DATA_COLUMNS, PARTITION_COLUMNS);
+        assertQuery("DESCRIBE stock_ticks_cow", sql);
+        assertQuery("DESCRIBE stock_ticks_mor_ro", sql);
+        assertQuery("DESCRIBE stock_ticks_mor_rt", sql);
+
+        // non-partition tables
+        @Language("SQL") String sql2 = generateDescribeIdenticalQuery(typeManager, HUDI_META_COLUMNS, DATA_COLUMNS, ImmutableList.of());
+        assertQuery("DESCRIBE stock_ticks_cown", sql2);
+        assertQuery("DESCRIBE stock_ticks_morn_ro", sql2);
+        assertQuery("DESCRIBE stock_ticks_morn_rt", sql2);
+    }
+
+    @Test
+    public void testDemoQuery1()
+    {
+        @Language("SQL") String sqlTemplate = "SELECT symbol, max(ts) FROM %s GROUP BY symbol HAVING symbol = 'GOOG'";
+        @Language("SQL") String sqlResult = "SELECT 'GOOG', '2018-08-31 10:59:00'";
+        @Language("SQL") String sqlResultReadOptimized = "SELECT 'GOOG', '2018-08-31 10:29:00'";
+
+        assertQuery(format(sqlTemplate, "stock_ticks_cow"), sqlResult);
+        assertQuery(format(sqlTemplate, "stock_ticks_cown"), sqlResult);
+        assertQuery(format(sqlTemplate, "stock_ticks_mor_ro"), sqlResultReadOptimized);
+        assertQuery(format(sqlTemplate, "stock_ticks_mor_rt"), sqlResult);
+        assertQuery(format(sqlTemplate, "stock_ticks_morn_ro"), sqlResultReadOptimized);
+        assertQuery(format(sqlTemplate, "stock_ticks_morn_rt"), sqlResult);
+    }
```

Comment:
Can we comment somewhere the structures of those tables? 
For other people, it seems hard to understand what we're testing against and can't verify if the expected the results are correct

Reply:
I added a doc `hudi-testing-data.md` to describe the testing data, and inserted a comment at the begining of `HudiTestingDataGenerator#generateData` to guide people to the doc.